karapace/schema_registry_apis.py

```diff
@@ -241,8 +227,27 @@ def send_kafka_message(self, key, value):
         future = self.producer.send(self.config["topic_name"], key=key, value=value)
         self.producer.flush(timeout=self.kafka_timeout)
         msg = future.get(self.kafka_timeout)
-        self.log.debug("Sent kafka msg key: %r, value: %r, offset: %r", key, value, msg.offset)
-        self.get_offset_from_queue(msg.offset)
+        sent_offset = msg.offset
+
+        self.log.info(
+            "Waiting for schema reader to caugth up. key: %r, value: %r, offset: %r",
+            key,
+            value,
+            sent_offset,
+        )
+
+        if self.ksr.offset_watcher.wait_for_offset(sent_offset, timeout=1 * MINUTE) is True:
+            self.log.info(
+                "Schema reader has found key. key: %r, value: %r, offset: %r",
+                key,
+                value,
+                sent_offset,
+            )
+        else:
+            raise RuntimeError(
```

Comment:
Should the error be domain specific here? And the response to client to be `504 Gateway timeout`?

Reply:
These are the current errors:

https://github.com/aiven/karapace/blob/60430a8eee3bbe8949dac3a43af9f623884333b9/karapace/schema_registry_apis.py#L18-L35

I tried to not change this ATM because those are compatible with CSR

Reply:
Internal error looks to be the behavior in CSR. Although there is an error code for operation timeout `50002`, e.g. https://docs.confluent.io/platform/current/schema-registry/develop/api.html#post--subjects-(string-%20subject)-versions